### ISSUE TITLE
'additionalProperties' breaks iOS codegen

### ISSUE THREAD

**Issue by samskiter** [swagger-api/swagger-codegen#1540]
additionalProperties breaks objective-c code generation as it causes generated classes to inherit from NSMutableDictionary, rather than from JSONModel. This would be fine but the generated class doesn't implement `initWithDictionary` either so the check inside SWGApiClient to allow it to be deserialised won't work:

```
   // model
    Class ModelClass = NSClassFromString(class);
    if ([ModelClass instancesRespondToSelector:@selector(initWithDictionary:error:)]) {
        return [[ModelClass alloc] initWithDictionary:data error:nil];
    }
```

**Comment by wing328** [comment]
@samskiter can you share the swagger spec us to repeat the issue ?

**Comment by samskiter** [comment]
hey, here's a schema I used:
```
{
    "id": "sitesGETresponse.schema.json#",
    "$schema": "http://json-schema.org/draft-04/schema#",
    "title": "Site list",
    "description": "Response to site list request",
    "type": "object",
    "properties": {
        "title": {
            "type": "string"
        }
    }, 
    "additionalProperties": {
        "type": "object",
        "properties": {
            "title": {
                "type": "string"
            },
            "count": {
                "type": "integer"
            },
            "uuid": {
                "type": "string"
            }
        },
        "required": ["uuid"]
    }
}
```

**Comment by samskiter** [comment]
(see also https://github.com/swagger-api/swagger-codegen/issues/453#issuecomment-154570630)

**Comment by wing328** [comment]
OK. It's caused by #1470 (map of inline model in this case)

**Comment by samskiter** [comment]
Ok, I'll take your word for it - on the face of it it looks like a different bug. So the generated code should implement initWithDictionary? And inherit from NSMutableDictionary as it currently does?

It might also be worth looking at supporting the recently introduced templates to objective-c. I think it's only available for arrays and dictionaries, but better than nothing ! :)

**Comment by wing328** [comment]
Btw, to workaround the issue for the time being, you can define the model properly in the `definitions` section of your spec.

**Comment by samskiter** [comment]
you mean like this?

```
{
    "id": "sitesGETresponse.schema.json#",
    "$schema": "http://json-schema.org/draft-04/schema#",
    "title": "Site list",
    "description": "Response to site list request",
    "type": "object",
    "properties": {
        "title": {
            "type": "string"
        }
    }, 
    "additionalProperties": {
        "$ref" : "otherfile.json"
}
```

Otherfile.json:

```
{
    "type": "object",
        "properties": {
            "title": {
                "type": "string"
            },
            "count": {
                "type": "integer"
            },
            "uuid": {
                "type": "string"
            }
        },
        "required": ["uuid"]
    }
}
```

**Comment by wing328** [comment]
Yes, that's one way to do it.

I was referring to https://github.com/swagger-api/swagger-spec/blob/master/examples/v2.0/json/petstore.json#L51 to define the models in the same file.

Models defined these ways have a proper name (instead of something like InlineSchema200)

**Comment by wing328** [comment]
@samskiter the issue has been fixed by #1734 in the latest master. Please pull the latest and let us know if you still see the issue.
